spec/core/string/slice_spec.rb

```diff
@@ -0,0 +1,453 @@
+  it_behaves_like :string_slice_regexp_index, :slice
+end
+
+quarantine! do
+  describe "String#slice with Regexp, group" do
+    it_behaves_like :string_slice_regexp_group, :slice
+  end
+end
+
+describe "String#slice with String" do
+  it_behaves_like :string_slice_string, :slice
+end
+
+describe "String#slice with Symbol" do
+  it_behaves_like :string_slice_symbol, :slice
+end
+
+describe "String#slice! with index" do
+  it "deletes and return the char at the given position" do
+    a = "hello"
+    a.slice!(1).should == ?e
+    a.should == "hllo"
+    a.slice!(-1).should == ?o
+    a.should == "hll"
+  end
+
+  it "returns nil if idx is outside of self" do
+    a = "hello"
+    a.slice!(20).should == nil
+    a.should == "hello"
+    a.slice!(-20).should == nil
+    a.should == "hello"
+  end
+
+  it "raises a FrozenError if self is frozen" do
+    -> { "hello".freeze.slice!(1)  }.should raise_error(FrozenError)
+    -> { "hello".freeze.slice!(10) }.should raise_error(FrozenError)
+    -> { "".freeze.slice!(0)       }.should raise_error(FrozenError)
+  end
+
+  it "calls to_int on index" do
+    "hello".slice!(0.5).should == ?h
+
+    obj = mock('1')
+    obj.should_receive(:to_int).at_least(1).and_return(1)
+    "hello".slice!(obj).should == ?e
+
+    obj = mock('1')
+    obj.should_receive(:respond_to?).at_least(1).with(:to_int, true).and_return(true)
+    obj.should_receive(:method_missing).at_least(1).with(:to_int).and_return(1)
+    "hello".slice!(obj).should == ?e
+  end
+
+
+  it "returns the character given by the character index" do
+    "hellö there".slice!(1).should == "e"
+    "hellö there".slice!(4).should == "ö"
+    "hellö there".slice!(6).should == "t"
+  end
+
+end
+
+describe "String#slice! with index, length" do
+  it "deletes and returns the substring at idx and the given length" do
+    a = "hello"
+    a.slice!(1, 2).should == "el"
+    a.should == "hlo"
+
+    a.slice!(1, 0).should == ""
+    a.should == "hlo"
+
+    a.slice!(-2, 4).should == "lo"
+    a.should == "h"
+  end
+
+  it "returns nil if the given position is out of self" do
+    a = "hello"
+    a.slice(10, 3).should == nil
+    a.should == "hello"
+
+    a.slice(-10, 20).should == nil
+    a.should == "hello"
+  end
+
+  it "returns nil if the length is negative" do
+    a = "hello"
+    a.slice(4, -3).should == nil
+    a.should == "hello"
+  end
+
+  it "raises a FrozenError if self is frozen" do
+    -> { "hello".freeze.slice!(1, 2)  }.should raise_error(FrozenError)
+    -> { "hello".freeze.slice!(10, 3) }.should raise_error(FrozenError)
+    -> { "hello".freeze.slice!(-10, 3)}.should raise_error(FrozenError)
+    -> { "hello".freeze.slice!(4, -3) }.should raise_error(FrozenError)
+    -> { "hello".freeze.slice!(10, 3) }.should raise_error(FrozenError)
+    -> { "hello".freeze.slice!(-10, 3)}.should raise_error(FrozenError)
+    -> { "hello".freeze.slice!(4, -3) }.should raise_error(FrozenError)
+  end
+
+  quarantine! do
```

Comment:
Honestly not sure why this one is failing. Something to do with method_missing not being called in the implicit conversion probably?